src/viam/sdk/resource/resource_server_base.hpp

```diff
@@ -8,14 +8,19 @@ namespace sdk {
 
 class ResourceServer {
    public:
-    virtual void register_server(std::shared_ptr<Server> server) = 0;
     const std::shared_ptr<ResourceManager>& resource_manager() const;
 
    protected:
-    ResourceServer(std::shared_ptr<ResourceManager> manager) : manager_(manager){};
+    ResourceServer(std::shared_ptr<ResourceManager> manager,
+                   std::shared_ptr<Server> server,
+                   grpc::Service* service)
+        : manager_(manager), server_(server) {
```

Comment:
- Please move the implementation out of line.
- `std::move` on `manager` and `server`.
- Have some doubts about this approach of having the `grpc::Service*` become part of the constructor signature for an API type.

Reply:
Can do the first two.

> Have some doubts about this approach of having the grpc::Service* become part of the constructor signature for an API type.

Did you have a different idea here?